File: tests/issue_11.py
#!/usr/bin/env python

import wave

##################################################

with wave.open("./issue-11.wav", "rb") as audio:

    print audio.getnframes()

##################################################

# <<< wave.Error: unknown format: 3 >>>

# Python's wave module works with a specific type of WAV: PCM (WAVE_FORMAT_PCM: 0x0001).
# In your case, you're using a WAV of type IEEE_FLOAT (WAVE_FORMAT_IEEE_FLOAT: 0x0003 = hex(3))

##################################################

File: tests/Issue11.hs
#!/usr/bin/env cabal

{- cabal:

  build-depends: base
               , wave

-}

--------------------------------------------------

{-# LANGUAGE PackageImports #-}

--------------------------------------------------

import qualified "wave" Codec.Audio.Wave as W

--------------------------------------------------

main = do

  putStrLn "--------------------------------------------------"
  putStrLn "Reading WAV (Unsigned PCM)"
  putStrLn "--------------------------------------------------"
  
  W.readWaveFile "./issue-11.wav.pcm"

  putStrLn "--------------------------------------------------"
  putStrLn "Reading WAV (F.P. PCM)"
  putStrLn "--------------------------------------------------"

  W.readWaveFile "./issue-11.wav"

  putStrLn "--------------------------------------------------"
  putStrLn "DONE"
  putStrLn "--------------------------------------------------"

--------------------------------------------------

-- https://github.com/mrkkrp/wave/issues/11
--
-- « readWaveFile » throws « NonPcmFormatButMissingFact » on « ./issue-11.wav ».
--

--------------------------------------------------